### ISSUE TITLE
warning error when configuring init-local.el

### ISSUE THREAD

**Issue by kirk86** [purcell/emacs.d#334]
Dear Steve,

apologies for the disturbance. I was wondering if I could ask you a couple of questions.
I've just downloaded your emacs.d configuration and used it with my emacs.
I am a total novice when it comes to configuration of emacs. 

From your instructions you've mentioned that we should put our configurations in our `~/.emacs.d./lisp/init-local.el` or `~/.emacs.d/init-preload-local.el` if we want our configurations to take effect upon starting emacs.

I've installed evil-mode using the package manager and here is what I've tried so far but without any luck.

1st attempt:
After installing evil, I entered the following line in the init.el
`(require 'evil)`
then 
I created a file with the name `init-evil.el` in `~/.emacs.d/lisp/` where I wrote `(evil-mode 1)`

This obviously didn't work since evil was not recognised upon starting emacs.

2nd attempt:
I left my initial configuration of `(require 'evil)` unchanged in `init.el`
and I simply created `~/.emacs.d/lisp/init-local.el` as in your notes and added
`(evil-mode 1)`
which also didn't had any effect but furthermore I received and error `error: Required feature init-local was not provided`

3rd attempt:
Left unchanged the configuration `(require 'evil)` in `init.el`
and I created `~/.emacs.d/lisp/init-preload-local.el` where I added `(evil-mode 1)`
which unfortunately gave the same error as before
`error: Required feature init-local was not provided`

Any hints on how to solve the above error.

If I want to add highlight row configuration where should I put it in `init.el` or `init-local.el`?

Finally, I am trying to setup emacs as an ide for:
```
python
latex-auctex
c/c++
java
matlab
```
would be kind enough if you have any previous experience to point me into the right direction?

IMHO, I think that the whole purpose of the package manager in emacs is overshadowed by the fact that even in the case were you would simply like the default settings of an installed package you will always have to resolve in adding the configuration in `init.el`  or wherever is appropriate.

In other words package manager in emacs doesn't take care of that which is a pitty.

Thanks in advance.

**Comment by purcell** [comment]
Hi - the error tells you what to do, and it's noted in the README too: you need a `(provide 'init-local)` form at the bottom of `init-local.el`. :-)

**Comment by kirk86** [comment]
Dear Steve,

I create both `init-local.el` and `init-preload-local.el` files and puted the following code in equivalently:

`1. init-local.el`

`;;; ============================================================
;;; Comentary:
;;; ============================================================

;;; ============================================================
;;; package --- Summary
;;; ============================================================
(require 'evil)
(evil-mode 1)


;;; =============================================================
;;; notify init.el where to find and load init-local.el
;;; =============================================================
(provide 'init-local)`


`2. init-preload-local.el`

`;;; ============================================================
;;; emacs configuration
;;; ============================================================

;; Highlight tabs and spaces
(setq-default highlight-tabs t)
(setq-default highlight-trailing-whitespace t)

;;change highlight colour
(set-face-attribute 'region t :background "#199540")

;;don't highlight the end of long lines
(setq whitespace-line-column 99999)

;; enable default global mode for line numbers
(global-linum-mode t)

;; add settings regarding the fill-column-indicator
;; enable fci-mode as a global minor mode
(define-globalized-minor-mode
  global-fci-mode fci-mode (lambda () (fci-mode 1)))
(global-fci-mode 1)
(setq fci-rule-widht 1)
(setq fci-rule-color "orange")

;;; =============================================================
;; notify init.el where to locate and load init-preload-local.el
;;; =============================================================
(provide 'init-preload-local)`

Now everytime that I open emacs a get the following message:

`The local variables list in /home/user/.emacs.d/lisp contains values that may not be safe (*)
Do you want to apply it y/n/!`

By googling around I found that this is something related to setting local variable values at the start of the source file. But that's all I know, I don't know how or why are used and I don't know if I have
screwed anything in the process from `init-local.el and init-preload-local.el` files during the customization.

How can I resolve this message issue?

Thank you.

**Comment by purcell** [comment]
Not sure, sorry. Those files look reasonable. The warning might be related to the `~/.emacs.d/lisp/.dir-locals.el` file, but the default version of that in my repo doesn't contain any variable values that may not be safe. Perhaps you've modified that file from the default?

**Comment by kirk86** [comment]
Hi Steve,

I checked the `~/.emacs.d/lisp.dir-locals.el` and this is the only thing that I've found writen on the file:
`((emacs-lisp-mode . ((no-byte-compile t))))`

is that the default value that you had on that file? Because I don't remember changing that file at all.

Any clues?

**Comment by purcell** [comment]
Yeah, that's the default. Sorry, then - it's not really obvious to me what the problem might be.

**Comment by kirk86** [comment]
Ok, thank you Steve!

**Comment by dcorking** [comment]
@kirk86 IIRC sometimes you will get that warning if you restore a saved emacs desktop, if that desktop contains a buffer that sets local variables.

**Comment by kirk86** [comment]
@dcorking Thank you David,

How can you tell the difference between a local and a global variable in the `~/.emacs`?
